src/main/java/seedu/duke/exercise/ExerciseList.java

```diff
@@ -0,0 +1,76 @@
+package seedu.duke.exercise;
+
+import java.util.ArrayList;
+
+public class ExerciseList {
+
+    public static final String MESSAGE_EXERCISE_ADDED = "An exercise has been added:";
+    private static final String MESSAGE_EXERCISE_DELETED = "You have removed the exercise:";
+    private static final String MESSAGE_EMPTY_EXERCISE_LIST = "No exercise is found.";
+    private ArrayList<Exercise> exerciseList = new ArrayList<>();
+
+    /**
+     * Returns exercise item at a specific index in the exercise list.
+     *
+     * @param index Index of the exercise item.
+     * @return exercise item in the exercise list.
+     */
+    public Exercise getExercise(int index) {
+        return this.exerciseList.get(index);
+    }
+
+    /**
+     * Returns the number of exercises in the exercise list.
+     *
+     * @return Number of exercises in the exercise list.
+     */
+    public int getSize() {
+        return exerciseList.size();
+    }
+
+    /**
+     * Adds an exercise item into the exercise list.
+     *
+     * @param exercise Exercise class object to be added.
+     */
+    public void addExercise(Exercise exercise) {
+        this.exerciseList.add(exercise);
+
+        System.out.println(MESSAGE_EXERCISE_ADDED);
+        System.out.println("\t" + exercise);
+    }
+
+    /**
+     * Deletes an exercise item from the exercise list.
+     *
+     * @param index Index of the exercise to be deleted.
+     */
+    public void deleteExercise(int index) {
+        Exercise deletedExercise = exerciseList.remove(index);
+
+        System.out.println(MESSAGE_EXERCISE_DELETED);
+        System.out.println("\t" + deletedExercise);
+        if (exerciseList.size() == 1) {
+            System.out.println("There is" + exerciseList.size() + "task left.");
+        } else {
+            System.out.println("There are" + exerciseList.size() + "tasks left.");
+        }
+    }
+
+    /**
+     * Prints out all the exercises in the exercise list.
+     */
+    public void printExerciseList() {
```

Comment:
Hmm for this method, I think we should pass the ArrayList to Ui to handle printing of lists instead of printing here

Reply:
Agreed, except that instead of passing it to UI as an ArrayList, we could format the list of exercises into a String here, and pass the String out.